File: jdr-final/src/main/monapp/entity/Plateau.java
package com.monapp.entity;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@Table(name="plateau")
public class Plateau {
	
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@JsonView(Views.Common.class)
	private int id;
	
	@OneToMany(mappedBy="plateau")
	@JsonView(Views.Plateau.class)
	private Set<Cellule> cellules = new HashSet<Cellule>();
	
	@OneToMany(mappedBy="plateau")
	@JsonView(Views.Plateau.class)
	private Set<Partie> parties;
	
	@Column
	@NotNull
	@JsonView(Views.Common.class)
	private int largeur;
	
	@Column
	@NotNull
	@JsonView(Views.Common.class)
	private int hauteur;
	
	public Plateau() {}
	
	public Plateau(Set<Cellule> cellules, Set<Partie> parties, int largeur, int hauteur) {
		this.cellules = cellules;
		this.parties = parties;
		this.largeur = largeur;
		this.hauteur = hauteur;
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public Set<Cellule> getCellules() {
		return cellules;
	}

	public void setCellules(Set<Cellule> cellules) {
		this.cellules = cellules;
	}

	public int getLargeur() {
		return largeur;
	}

	public void setLargeur(int largeur) {
		this.largeur = largeur;
	}

	public int getHauteur() {
		return hauteur;
	}

	public void setHauteur(int hauteur) {
		this.hauteur = hauteur;
	}

	public Set<Partie> getParties() {
		return parties;
	}

	public void setParties(Set<Partie> parties) {
		this.parties = parties;
	}
	
	
	
	
	
}

File: app/plateau.ts
import {Cellule} from './cellule';

export class Plateau {
  constructor(
    private _id: number,
    private _cellules: Array<Cellule>,
    private _hauteur: number,
    private _largeur: number
  ) {}


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get cellules(): Array<Cellule> {
    return this._cellules;
  }

  set cellules(value: Array<Cellule>) {
    this._cellules = value;
  }

  get hauteur(): number {
    return this._hauteur;
  }

  set hauteur(value: number) {
    this._hauteur = value;
  }

  get largeur(): number {
    return this._largeur;
  }

  set largeur(value: number) {
    this._largeur = value;
  }
}
